python/tk_multi_launchapp/software_entity_launcher.py

```diff
@@ -0,0 +1,196 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+import pprint
+
+import sgtk
+
+from .base_launcher import BaseLauncher
+
+shotgun_data = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_data")
+
+class SoftwareEntityLauncher(BaseLauncher):
+    """
+    Launches a DCC application based on site Software entity entries.
+    """
+    def register_launch_commands(self):
+        """
+        Determine what launch command(s) to register with the current TK engine.
+        Multiple commands may be registered based on the number of retrieved
+        Software entities and their corresponding 'versions' field.
+        """
+        # Determine the information to retrieve from Shotgun
+        # @todo: The 'sg_software_entity' setting can be removed once the
+        #        Software entity becomes native.
+        sw_entity = self._tk_app.get_setting("sg_software_entity") or "Software"
+
+        # Use filters to retrieve Software entities that match specified
```

Comment:
As discussed, the filter is being procedurally generated to account for empty Project and/or User settings in the current context.  Otherwise, the suggested filter works great!